sectionsUtilityTool/src/main/webapp/assets/js/utils.js

```diff
@@ -97,38 +116,38 @@ $('#helpLink').click(function(){
 
 //handler for the Update Course button
 $(document).on('click', '.setSections', function (e) {
+  var server = $('#serverInfo').text();
+  $('#postXListDone').hide();
+  $('#postXList').show();
+  $('#xListConfirm').hide();
   e.preventDefault();
   $('#debugPanel').empty();
   var thisCourse = $(this).attr('data-courseid');
 
-  var thisCourseTitle = $(this).closest('li').find('a.courseLink').text();
+  var thisCourseContainer = $(this).closest('li.course');
+  var thisCourseTitle = thisCourseContainer.find('a.courseLink').text();
+
   var $sections = $(this).closest('li').find('ul').find('li');
+  
   var posts = [];
   $('#xListInner').empty();
   $('#xListInner').append('<p><strong>' + thisCourseTitle + '</strong></p><ol id="listOfSectionsToCrossList" class="listOfSectionsToCrossList"></ol>');
+
+  $('#xListConfirm').attr('href',server + '/courses/' + thisCourse + '/settings#tab-sections');
   $sections.each(function( ) {
     posts.push('/api/v1/sections/' + $(this).attr('data-sectionid') + '/crosslist/' + thisCourse);
     $('#listOfSectionsToCrossList').append( '<li id=\"xListSection' + $(this).attr('data-sectionid') + '\">' + $(this).find('div.sectionName span').text() + '</li>');
   });
   $('#postXList').click(function(){
-    var index, len;
-    for (index = 0, len = posts.length; index < len; ++index) {
-      var xListUrl ='manager' + posts[index];
-      $.post(xListUrl, function(data) {
-        var section = data.id;
-        $('#xListSection' +  section).append('<span class=\"label label-success\">Success</span>');
-      })
-      .fail(function(data) {
-        var section = data.id;
-        $('#xListSection' +  section).append('<span class=\"label label-failure\">Failure</span>');
-      })
-      .always(function() {
-        // need to count the success / failures
-      });
-
-
-    }  
-  });
+    var xListPosts = doXListPosts(posts);
+    $.when.apply($, xListPosts).done(function() {
```

Comment:
In this anonymous success function, you can check the contents of `xListPostStatus.successes` and `xListPostStatus.failures`, then do something useful.  If those arrays contain the data of the requests, you might be able to get to some useful information easily.  

On the other hand, if you don't want to (or run into trouble with) tracking these successes and failures, you could use `$('.label-failure')` to look for elements that contain failures, but you may need to parse the information you need out of the elements.  Keeping a list is probably better, since you can skip the selecting, searching, parsing, etc.

Of course, you can make my example a little simpler by tracking only the failures.